### PR TITLE
chore: update to tree-sitter 22.6

### LINKED ISSUE #4: Update tree-sitter version
Thanks for creating and maintaining a better TS parser for PowerShell. My use case needs the most recent version of tree-sitter, and the deps for this are a couple versions behind.

### PR DESCRIPTION
**Pull request by acarl005**
This PR closes #4 and updates tree-sitter to the most recent versions. Most of this diff is auto-generated code, and I'll highlight my actual changes with inline remarks.

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -217,7 +217,7 @@ module.exports = grammar({
       $.braced_variable
     ),
 
-    braced_variable: $=> /\${[^}]+}/,
+    braced_variable: $=> /\$\{[^}]+\}/,
```
Comment (acarl005) [inline grammar.js:220]: I had to manually escape these according to [this change](https://github.com/tree-sitter/tree-sitter/pull/2838)

File: package.json [hunk 2]
```diff
@@ -3,22 +3,35 @@
...
   "main": "bindings/node",
+  "types": "bindings/node",
   "scripts": {
     "generate": "tree-sitter generate",
     "build-node": "tree-sitter generate && node-gyp build",
     "test": "tree-sitter test",
     "parse": "tree-sitter parse",
     "parse-debug": "tree-sitter parse -d",
-    "build-wasm":  "tree-sitter build-wasm"
+    "build-wasm": "tree-sitter build --wasm",
```
Comment (acarl005) [inline package.json:13]: The `build-wasm` subcommand is deprecated.

File: package.json [hunk 3]
```diff
@@ -3,22 +3,35 @@
...
+  },
+  "peerDependenciesMeta": {
+    "tree_sitter": {
+      "optional": true
+    }
   },
   "devDependencies": {
     "node-gyp": "^9.4.0",
-    "tree-sitter-cli": "^0.20.8"
+    "tree-sitter-cli": "^0.22.6",
```
Comment (acarl005) [inline package.json:33]: I bumped this version string.

File: package.json [hunk 4]
```diff
@@ -32,5 +45,13 @@
...
       ]
     }
+  ],
+  "files": [
+    "grammar.js",
+    "binding.gyp",
+    "prebuilds/**",
+    "bindings/node/*",
+    "queries/*",
+    "src/**"
```
Comment (acarl005) [inline package.json:55]: The rest of this file had automated changes by.. I'm not sure what exactly. I'm not too familiar with node development and so I was a little surprised to see anything in `package.json` get changed apart from those 2 lines I edited.

**Comment by citronneur** [comment]
Thanks for your contribution, I had to fix some changed in the test process but it works with the last version of tree-sitter ! Thanks again

**Pull request closed without merging**